ADAL/src/request/ADAuthenticationRequest+Broker.m

```diff
@@ -190,62 +189,69 @@ + (ADAuthenticationResult *)processBrokerResponse:(NSURL *)response
     NSString *qp = [components percentEncodedQuery];
     //expect to either response or error and description, AND correlation_id AND hash.
     NSDictionary* queryParamsMap = [NSDictionary adURLFormDecode:qp];
-    
+
+    NSString* keychainGroup = resumeDictionary[@"keychain_group"];
+
     if([queryParamsMap valueForKey:OAUTH2_ERROR_DESCRIPTION])
     {
-        return [ADAuthenticationResult resultFromBrokerResponse:queryParamsMap];
+        // In the case where Intune App Protection Policies are required, the broker may send back the Intune MAM Resource token
+        NSMutableDictionary* userInfo = [[NSMutableDictionary alloc] initWithDictionary:queryParamsMap];
+        if (queryParamsMap[BROKER_INTUNE_HASH_KEY] && queryParamsMap[BROKER_INTUNE_RESPONSE_KEY])
+        {
+            ADAuthenticationError* intuneTokenError = nil;
+            NSDictionary* intuneTokenResponse = [ADHelpers decryptBrokerResponse:@{BROKER_RESPONSE_KEY:queryParamsMap[BROKER_INTUNE_RESPONSE_KEY],
+                                                                                   BROKER_HASH_KEY:queryParamsMap[BROKER_INTUNE_HASH_KEY],
+                                                                                   BROKER_MESSAGE_VERSION:queryParamsMap[BROKER_MESSAGE_VERSION] ? queryParamsMap[BROKER_MESSAGE_VERSION] : @1}
+                                                                   correlationId:correlationId
+                                                                           error:&intuneTokenError];
+
+            ADAuthenticationResult* intuneTokenResult = [[ADTokenCacheItem new] processTokenResponse:intuneTokenResponse
+                                                                                         fromRefresh:NO
+                                                                                requestCorrelationId:intuneTokenResponse[OAUTH2_CORRELATION_ID_RESPONSE]];
+
+            if (!resumeDictionary || !keychainGroup)
+            {
+                AD_LOG_WARN(correlationId, @"Failed to cache Intune token, no resume state found in NSUserDefaults");
```

Comment:
Don't we already check for resumeDictionary existence before processing the whole broker response?